File: core/java/src/main/java/hara/lang/protocol/IPointer.java
package hara.lang.protocol;

import hara.lang.declaration.HaraMethod;
import hara.lang.declaration.HaraProtocolBinding;

@HaraProtocolBinding(namespace = "std.protocol.ipointer", name = "IPointer")
public interface IPointer {
  @HaraMethod(value = "ptr-context", arity = 1)
  Object ptrContext();
}

File: core/rust/src/lang/protocol/ipointer.rs
use hara_protocol_macros::hara_protocol;

/// Identifies the evaluator context that owns pointer resolution.
///
/// Pointer descriptor fields are accessed through the ordinary collection
/// protocols; they are deliberately not duplicated here.
#[hara_protocol(namespace = "std.protocol.ipointer", name = "IPointer")]
pub trait IPointer<C> {
    #[hara_method(value = "ptr-context", arity = 1)]
    fn pointer_context(&self) -> C;
}
